scripts/bin/revbump

```diff
@@ -1,95 +1,99 @@
-		echo "${package}: skipping as no build.sh found"
-		continue
-	fi
-
-	if grep -qP '^TERMUX_PKG_REVISION=(\d+)$' "${buildsh_path}"; then
-		# Increment revision.
-		current_rev=$(. "${buildsh_path}" 2>/dev/null; echo "${TERMUX_PKG_REVISION}")
-		incremented_rev=$((current_rev + 1))
-
-		# Replace revision value in build.sh.
-		if ! ${IS_GIT_REPOSITORY}; then
-			echo "${package}: bumping to ${incremented_rev}"
-		fi
-		sed -i -E "s/^(TERMUX_PKG_REVISION=)(.*)$/\1${incremented_rev}/g" "${buildsh_path}"
-	else
-		# Add base (1) revision right after TERMUX_PKG_VERSION.
-		if ! ${IS_GIT_REPOSITORY}; then
-			echo "${package}: setting revision to 1"
-		fi
-		sed -i -E "/TERMUX_PKG_VERSION=(.*)/a TERMUX_PKG_REVISION=1" "${buildsh_path}"
-	fi
-
-	# If we are on Git repository, prompt for committing changes.
-	if ${IS_GIT_REPOSITORY}; then
-		echo
-		echo "You are about to commit these changes:"
-		echo
-		echo "--------------------"
-		git --no-pager diff --patch "${buildsh_path}"
-		echo "--------------------"
-		echo
-		echo "rebuild(${repo}/${package}): $1"
-		echo
-		read -re -p "Do you want to commit these changes ? (y/n) " CHOICE
-		echo
-		if [[ ${CHOICE} =~ (Y|y) ]]; then
-			git add "${buildsh_path}"
-			git commit -m "rebuild(${repo}/${package}): $1"
-		else
-			echo "Not committing to Git!"
-		fi
-		echo
-	fi
-done
+#!/usr/bin/env python3
+import os
+import json
+import sys
+
+filename = sys.argv[0].split("/")[-1]
+
+usage = f"""
+Usage:
+
+{filename} <package> ...            
+
+Add or increment TERMUX_PKG_REVISION of package. 
+
+
+{filename} --dependencies  <package>    
+
+Add or increment TERMUX_PKG_REVISION of all the packages that have <package> in TERMUX_PKG_DEPENDS or in TERMUX_PKG_BUILD_DEPENDS
+"""
+
+
+FOLDERS = list(json.load(open("repo.json")).keys())
+
+
+def get_build_dependent_files(folders: list, DEP : str) -> list:
+    """
+    Gets all the packages that depend on some package
+    """
+    build_files = []
+    for d in folders:
+        for folder in os.listdir(d):
+            for line in open(d+"/"+folder+"/build.sh","r").read().split("\n"):
+                if line.startswith("TERMUX_PKG_DEPENDS") or line.startswith("TERMUX_PKG_BUILD_DEPENDS"):
+                    if DEP in line:
+                        build_files.append(d+"/"+folder+"/build.sh")
+    return build_files
+
+
+def bump_revision(file:str) -> None:
+    """ 
+    Bumps version in file
+    """
+    file_read = open(file,"r").read().split("\n")
+    revision = 0
+    data = ""
+
+    if "TERMUX_PKG_REVISION" in "\n".join(file_read):
+        for line in file_read:
+            if line.startswith("TERMUX_PKG_REVISION"):
+                revision = line.split("=")[-1]
+                if revision.startswith('"'):
+                    revision = revision[1]
+                revision = int(revision) + 1
+                file_read[file_read.index(line)] = line.split("=")[0]+"="+str(revision)
+                open(file,"w").write("\n".join(file_read))
```

Comment:
I don't think it is a good idea to open the same file twice, for reading and writing, in parallel. Close the file when the respective operations are done